linker/shared/src/test/scala/org/scalajs/linker/checker/ClassDefCheckerTest.scala

```diff
@@ -479,6 +479,11 @@ class ClassDefCheckerTest {
       superCtorCall
     )
 
+    testRestrictedThisError(
+      Assign(Select(thiz, FieldName("Bar", "y"))(IntType), int(5)),
```

Comment:
Maybe this is nit-picky, but can we make sure the field in question is actually defined? Otherwise we might end up with an unnecessarily brittle test.

Reply:
We can't. It's a `ClassDefChecker` test. It does not even *see* the superclass.

Reply:
:facepalm: indeed